pkg/kv/kvserver/replica_write.go

```diff
@@ -171,7 +174,8 @@ func (r *Replica) executeWriteBatch(
 
 	// If the command is proposed to Raft, ownership of and responsibility for
 	// the concurrency guard will be assumed by Raft, so provide the guard to
-	// evalAndPropose.
+	// evalAndPropose. If we return with an error from executeWriteBatch, the
+	// caller reassumes ownership over the Guard.
```

Comment:
This confused me a bit, since we return errors after moving ownership to Raft as well.


```suggestion
	// evalAndPropose. If we return with an error from executeWriteBatch, we
  // also return the guard which the caller reassumes ownership of.
```